x-pack/solutions/security/plugins/security_solution/scripts/telemetry/build_ebt_data_view.test.ts

```diff
@@ -0,0 +1,377 @@
+            obj: {
+              properties: {
+                b: { type: 'long' },
+                c: { type: 'text' },
+                d: {
+                  properties: {
+                    e: { type: 'date' },
+                  },
+                },
+              },
+            },
+          },
+        },
+        lone: { type: 'ip' },
+      } as any;
+
+      const out = flattenSchema(schema);
+
+      expect(out).toEqual({
+        'root.a': 'keyword',
+        'root.obj.b': 'long',
+        'root.obj.c': 'text',
+        'root.obj.d.e': 'date',
+        lone: 'ip',
+      });
+    });
+
+    test('ignores non-object or null schema nodes', () => {
+      const schema = {
+        a: null,
+        b: 42,
+        c: 'str',
+        d: undefined,
+      } as any;
+
+      const out = flattenSchema(schema);
+
+      expect(out).toEqual({});
+    });
+
+    test('array of primitive types maps to item type', () => {
+      const schema = {
+        tags: { type: 'array', items: { type: 'keyword' } },
+        counts: { type: 'array', items: { type: 'long' } },
+        dates: { type: 'array', items: { type: 'date' } },
+      } as any;
+
+      const out = flattenSchema(schema);
+
+      expect(out).toEqual({
+        tags: 'keyword',
+        counts: 'long',
+        dates: 'date',
+      });
+    });
+
+    test('array of objects flattens item properties with parent prefix', () => {
+      const schema = {
+        rules: {
+          type: 'array',
+          items: {
+            properties: {
+              name: { type: 'keyword' },
+              score: { type: 'integer' },
+              meta: {
+                properties: {
+                  enabled: { type: 'boolean' },
+                },
+              },
+            },
+          },
+        },
+      } as any;
+
+      const out = flattenSchema(schema);
+
+      expect(out).toEqual({
+        'rules.name': 'keyword',
+        'rules.score': 'integer',
+        'rules.meta.enabled': 'boolean',
+      });
+    });
+
+    test('array without items is emitted as array', () => {
+      const schema = {
+        unknown: { type: 'array' },
+      } as any;
+
+      const out = flattenSchema(schema);
+
+      expect(out).toEqual({
+        unknown: 'array',
+      });
+    });
+
+    test('array with unknown item shape is emitted as array', () => {
+      const schema = {
+        misc: { type: 'array', items: {} },
+      } as any;
+
+      const out = flattenSchema(schema);
+
+      expect(out).toEqual({
+        misc: 'array',
+      });
+    });
+
+    test('nested arrays: array of objects containing array of primitives', () => {
+      const schema = {
+        groups: {
+          type: 'array',
+          items: {
+            properties: {
+              ids: { type: 'array', items: { type: 'long' } },
+              labels: { type: 'array', items: { type: 'keyword' } },
+            },
+          },
+        },
+      } as any;
```

Comment:
Instead of `any` it's possible to use `@ts-expect-error` above `const out = flattenSchema(schema);` line.